### PR TITLE
Feat(eos_cli_config_gen): Add support for flow tracking hardware

### PR DESCRIPTION
**Pull request by gmuloc**
## Change Summary

Add support for `flow tracking hardware`

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes

Following discussion - introduce new top level `flow_tracking` - deprecates `flow_trackings` one as the different types don't have the same keys despite what the cli show command display.

## How to test

molecule test

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/eos_cli_config_gen.schema.yml [hunk 1]
```diff
@@ -2480,6 +2480,7 @@ keys:
           type: str
           valid_values:
           - sampled
+          - hardware
           description: Flow Tracking Type - only 'sampled' supported for now
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/eos_cli_config_gen.schema.yml]: The description should be updated. No need to write the possible options. But maybe say that supported values are hardware dependent like we do elsewhere.

**Comment by gmuloc** [comment]
This PR does not allow to configure `flow tracker hardware FOO` under ethernet-interfaces. This will be tracked in another issue.

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/flow-tracking.md [hunk 2]
```diff
@@ -91,6 +91,30 @@ flow tracking sampled
          collector dead:beef::cafe
       flow table size 100000 entries
    no shutdown
+!
+flow tracking hardware
```
Comment (carlbuchmann) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/flow-tracking.md:115]: All commands are accepted but not all commands are retained on the cEOS-lab dut running 4.30.3M.
Note that this maybe a false positive due to testing on cEOS-lab.

i.e. `sample 666`, `record export mpls`.

```shell
eos-1(config-flow-tracking-hardware)#sh active 
flow tracking hardware
   tracker T1
      record export on inactive timeout 3666
      record export on interval 5666
   tracker T2
      exporter T2-E1
         collector 42.42.42.42
   tracker T3
      exporter T3-E1
      exporter T3-E2
         collector 10.10.10.10 port 777
      exporter T3-E3
         format ipfix version 10
         collector this.is.my.awesome.collector.dns.name port 888
         local interface Management1
         template interval 424242
      exporter T3-E4
         collector dead:beef::cafe
   no shutdown
eos-1(config-flow-tracking-hardware)#
```


Slightly diffrent behavior on sampled:
```
eos-1(config-flow-tracking-sampled)#sh active 
flow tracking sampled
   tracker T1
      record export on inactive timeout 3666
      record export on interval 5666
      record export mpls
   tracker T2
      flow table size 614400 entries
      exporter T2-E1
         collector 42.42.42.42
   tracker T3
      flow table size 100000 entries
      exporter T3-E1
      exporter T3-E2
         collector 10.10.10.10 port 777
      exporter T3-E3
         format ipfix version 10
         collector this.is.my.awesome.collector.dns.name port 888
         local interface Management1
         template interval 424242
      exporter T3-E4
         collector dead:beef::cafe
eos-1(config-flow-tracking-sampled)#
```

**Review by carlbuchmann — DISMISSED** [review]
LGTM - commands being retained may not be something we need to worry about as this may be platform specific support.

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/flow-tracking.md (hunk 2 above)
Comment (carlbuchmann) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/flow-tracking.md:115]: Let me test this on a CloudEOS node

Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/flow-tracking.md:115]: We should check with a `show active all` under the context.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/flow_trackings.schema.yml [hunk 3]
```diff
@@ -14,8 +14,8 @@ keys:
       keys:
```
Comment (carlbuchmann) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/flow_trackings.schema.yml]: Let's create a new key called `flow_tracking` with dictionary items: `sampled` and `hardware`.
We will deprecate flow_tracking and remove it in 5.0.0

Comment (gmuloc) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/flow_trackings.schema.yml]: This would clash with existing `flow_tracking` top level key - maybe you made a typo in this comment?

Comment (gmuloc) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/flow_trackings.schema.yml]: ok deprecated `flow_trackings` - this PR is NOT anymore just adding a type :p 

e1d9085

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/flow-trackings.j2 [hunk 4]
```diff
@@ -6,7 +6,7 @@
 {# doc - flow trackings #}
 {% if flow_trackings is arista.avd.defined %}
 
-### Flow Tracking
+### Flow Tracking - Deprecated
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/flow-trackings.j2]: @ClausHolbechArista let me know if this is "ok" - ideally the two models should never be used together

Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/flow-trackings.j2]: Normally we just produce docs from both models in the deprecation period. Under the same tables.

Comment (gmuloc) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/flow-trackings.j2]: Done in [88aefb2](https://github.com/aristanetworks/ansible-avd/pull/3199/commits/88aefb228992739e060e7258198e53d15deae97c)

**Review by ClausHolbechArista — DISMISSED** [review]

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/flow_trackings.schema.yml [hunk 5]
```diff
@@ -7,15 +7,19 @@
...
     type: list
     primary_key: type
     items:
       type: dict
       keys:
         type:
           type: str
-          valid_values: ['sampled']
-          description: Flow Tracking Type - only 'sampled' supported for now
+          valid_values: ['sampled', 'hardware']
```
Comment (carlbuchmann) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/flow_trackings.schema.yml]: We should revert these changes, and only "sampled" should be supported in the old model.

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/flow-tracking.yml [hunk 6]
```diff
@@ -64,3 +104,4 @@ port_channel_interfaces:
   - name: Port-Channel42
     flow_tracker:
       sampled: T3
+      hardware: T4
```
Comment (carlbuchmann) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/flow-tracking.yml]: invalid tracker name so we don't see it in the documentation report table, will update to T3

Comment (carlbuchmann) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/flow-tracking.yml]: resolved in [20c9a42](https://github.com/aristanetworks/ansible-avd/pull/3199/commits/20c9a42ef78c021d98c21597c080946e9d976c7b)

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/flow_trackings.schema.yml (hunk 5 above)
Comment (carlbuchmann) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/flow_trackings.schema.yml]: resolved in [713d986](https://github.com/aristanetworks/ansible-avd/pull/3199/commits/713d986fbe61d087620d72d5591bc7d02ca88c2f)

**Review by carlbuchmann — APPROVED** [review]
LGTM

**Review by ClausHolbechArista — APPROVED** [review]
